### ISSUE TITLE
Check DOI Reassign workflow in OJS 3.2

### ISSUE THREAD

**Issue by jmacgreg** [pkp/pkp-lib#5515]
**Expected behaviour:** this button should clear and recreate DOIs that have already been assigned, including published DOIs. 

**How to check:** 
1. publish some articles, and make sure DOIs are assigned. Also have some articles that are scheduled but not yet published, but with DOIs assigned. 
2. Go to Settings > Website > Plugins > DOI Plugin, change the DOI pattern behaviour, and save. 
3. In the DOI Plugin settings, click the Reassign DOIs button. 
4. Check the published articles: their DOIs should now reflect the new DOI pattern behaviour. 
5. Check the scheduled but unpublished articles: their DOIs should now reflect the new DOI pattern behaviour. 

(NOTE: @AhemNason and I agree that this is brute behaviour, but it's a brute button. In future we'd be very comfortable having the reassign option relegated to a command-line option only, or otherwise removed from day-to-day availability.)

**Comment by asmecher** [comment]
Just checking, are you or @AhemNason testing this for the 3.2 release, or looking for confirmation from the dev team?

**Comment by jmacgreg** [comment]
We'll test and confirm - @AhemNason, sounds good?

**Comment by AhemNason** [comment]
yes!

**Comment by AhemNason** [comment]
So, Reassign DOIs currently wipes all DOIs. Can confirm that that part works great. I still think the copy here is super unintuitive. 

> If you change your DOI configuration, DOIs that have already been assigned will not be affected. Once the DOI configuration is saved, use this button to clear all existing DOIs so that the new settings will take effect with existing objects.

Currently, the behaviour is that it wipes all DOI suffixes across the whole journal. You can then click the "assign dois" button to assign DOIs based on the suffix pattern you've selected. But, the button doesn't "reassign" anything. It also wiped DOIs that were assigned. My list of DOIs went from populated (with some assigned) to fully empty. 

If a DOI has a status of "active", you shouldn't be able to wipe it.

**Comment by NateWr** [comment]
> I still think the copy here is super unintuitive

We really just need someone to provide better copy. Can you do that for us?

**Comment by NateWr** [comment]
Or maybe the docs team has already written clearer documentation on this from which copy can be crafted? @amandastevens do you know?

**Comment by amandastevens** [comment]
We haven't tackled this yet.  I didn't know this was a new feature of 3.2.

**Comment by NateWr** [comment]
I don't think this is new but no worries. Just checking to see if any existing docs had been written up. :+1:

**Comment by AhemNason** [comment]
I'll try to spell this all out clearly based on what's currently happening (and we can figure out if that's what we want to happen) 

_**What it currently says**_
> **Reassign DOIs**
> If you change your DOI configuration, DOIs that have already been assigned will not be affected. Once the DOI configuration is saved, use this button to clear all existing DOIs so that the new settings will take effect with existing objects.

_**What it's actually doing**_
If you choose to "reassign", currently, it will wipe the DOI suffix from all article metadata recorded in the database. Regardless of whether or not it's been assigned. Essentially, it wipes all DOIs entirely. 

_**How I'd rephrase it**_
> **Clear All DOIs**
> This will clear all DOIs for this journal and allow you to reassign them using the "Assign DOIs" button. Please note, it is not recommended to do this if you have any custom DOIs. 

----

Recommendations: 

1. Clearing all DOIs in this way should be a command-line function. It is "the nuclear option". At the very least, it should have a modal that asks if folks are sure. 
2. Previous behaviour sort of combined the current "reassign" and "assign" functions. It would automatically assign/overwrite all DOI suffixes for articles to match the pattern defined in the DOI plugin. 
3. The text here says it won't "DOIs that have already been assigned will not be affected." The assignment isn't really the issue so much as registration. I don't know if we could have the Crossref and Datacite plugins reporting back to the DOI plugin to say "this DOI is active, don't change it". Or, maybe it should be "DOIs for published articles will not be affected". It really depends on what we want the features here to look like.

**Comment by amandastevens** [comment]
It would be great to add something about this feature to the [DOI Plugin Guide](https://docs.pkp.sfu.ca/doi-plugin/en/) after we clarify how it works.

**Comment by AhemNason** [comment]
Yeah, actually. That's a very good point @amandastevens. This version of OJS changes DOI behaviour in a not-insignificant way. I can start drafting a 3.2 version of that this week and have it for release. I can use the testing install for screenshots since I'm testing it already.

**Comment by asmecher** [comment]
Just pinging this issue to make sure anything necessary for 3.2 is in progress for Friday!

**Comment by AhemNason** [comment]
Tested and working consistently. I still think a UX test would be a good call for the process but nothing seems to be broken here!

**Comment by asmecher** [comment]
Excellent, thanks, @AhemNason!
